kernel/src/main/java/org/kframework/parser/json/JsonParser.java

```diff
@@ -65,6 +65,7 @@ public class JsonParser {
                              , KNONTERMINAL         = "KNonTerminal"
                              , KMODULE              = "KModule"
                              , KFLATMODULE          = "KFlatModule"
+                             , KFLATMODULELIST      = "KFlatModuleList"
                              , KIMPORT              = "KImport"
                              , KPRODUCTION          = "KProduction"
                              , KREGEXTERMINAL       = "KRegexTerminal"
```

Comment:
I don't think this is necessary is it? Can't we just use a `KDefinition` here instead?

Reply:
Or, if we do it this way, then both the Java code and the Python code should be modified so that a `KDefinition` contains a `KFlatModuleList` instead of a `List[KFlatModule]`.

But I still think we should just use a `KDefinition` here, because then we get all the facilities that `KDefinition` has (like `syntax_productions` and `functions` methods).

Reply:
The problem with using a `KDefinition` is that it recursively serializes all included modules, along with ensuring other soundness checks.
For this particular case, we want to serialize only the modules included by the main-spec-module but not by the main-def-module. So `KFlatModuleList` is just that, a list of modules.
We do a similar thing in Java in kprove: https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/kprovex/ProofDefinitionBuilder.java#L51
I had to introduce the new name because of how the serializer and deserializer is built. At each node in the AST I need to make a decision about how to deserialize the rest of the term.